### PR TITLE
Improvements to TextField and Picker components

### PR DESCRIPTION
**Pull request by dhruska**
- Allow text style props
- Add `placeholderTextColor` prop to Picker
- Get margins and paddings working correctly for Picker

### CODE REVIEW

File: packages/core/src/utilities.ts [hunk 1]
```diff
@@ -32,3 +32,25 @@ export function extractStyles(style: StyleProp<any>) {
 
   return { viewStyles, textStyles };
 }
+
+/**
+ * Merges a style object on top of another style object. In React Native,
+ * keys with undefined values in a style object will still override styles
+ * that appear earlier in a sequence. This avoids that problem.
+ */
+export function applyStyles(
```
Comment (dhruska) [inline packages/core/src/utilities.ts:41]: This lets us get rid of that `...(something ? { something } : {})` pattern.

As discussed in a previous PR, there doesn't seem to be a better way to do this. These all seem to _not_ work (i.e. they all result in `{ color: undefined }`

- `const mergedStyles = [{ color: "red" }, { color: undefined }]`
- `const mergedStyles = StyleSheet.compose({ color: "red" }, { color: undefined })`
- `const mergedStyles = StyleSheet.flatten([{ color: "red" }, { color: undefined }])`

File: packages/core/src/components/TextField.tsx [hunk 2]
```diff
@@ -427,27 +436,22 @@ class TextField extends React.Component<Props, State> {
...
-                  ...(borderWidth ? { borderWidth } : {}),
-                  ...(borderTopWidth ? { borderTopWidth } : {}),
-                  ...(borderRightWidth ? { borderRightWidth } : {}),
-                  ...(borderBottomWidth ? { borderBottomWidth } : {}),
-                  ...(borderLeftWidth ? { borderLeftWidth } : {}),
-                  ...(borderCol ? { borderColor: borderCol } : {}),
-                }
-              : {},
-          ]}
+          style={applyStyles(StyleSheet.flatten(containerStyle), {
```
Comment (peterpme) [inline packages/core/src/components/TextField.tsx:439]: do you think we can include the flatten inside applyStyles?

Comment (dhruska) [reply packages/core/src/components/TextField.tsx]: Definitely - made this change

File: packages/core/src/utilities.ts (hunk 1 above)
Comment (peterpme) [reply packages/core/src/utilities.ts:41]: do you mind putting this as a comment in there as an example? i think it'll make more sense

Comment (dhruska) [reply packages/core/src/utilities.ts:41]: 👍 sure thing

File: packages/core/src/utilities.ts [hunk 3]
```diff
@@ -32,3 +32,27 @@ export function extractStyles(style: StyleProp<any>) {
...
+  baseStyles: StyleProp<any>,
+  stylesToApply: StyleProp<any> | undefined
+) {
+  if (!stylesToApply) {
+    return;
+  }
+
+  const flattenedStyles = StyleSheet.flatten(baseStyles);
```
Comment (peterpme) [inline packages/core/src/utilities.ts:49]: will the base styles always be unflattened? for example, if i run stylesheet.flatten outside of this function?

do you think we should add a check to see if its already flat? maybe this part isn't possible, but i could have sworn styles returns a number, and once flattened, returns an object

**Review by adnelson — APPROVED** [review]

File: packages/core/src/utilities.ts [hunk 4]
```diff
@@ -32,3 +32,27 @@ export function extractStyles(style: StyleProp<any>) {
 
   return { viewStyles, textStyles };
 }
+
+/**
+ * Merges a style object on top of another style object. In React Native,
+ * keys with undefined values in a style object will still override styles
+ * that appear earlier in a sequence. This avoids that problem.
+ */
```
Comment (dhruska) [inline packages/core/src/utilities.ts:40]:
```suggestion
/**
 * Merges a style object on top of another style object. In React Native,
 * keys with undefined values in a style object will still override styles
 * that appear earlier in a sequence. This avoids that problem.
 *
 * This lets us avoid the `...(something ? { something } : {})` pattern.
 * There doesn't seem to be a better way to do this. These all seem to not
 * work (i.e. they all result in `{ color: undefined }`:
 * const mergedStyles = `[{ color: "red" }, { color: undefined }]`
 * const mergedStyles = `StyleSheet.compose({ color: "red" }, { color: undefined })`
 * const mergedStyles = `StyleSheet.flatten([{ color: "red" }, { color: undefined }])`
 */
```

File: packages/core/src/utilities.ts (hunk 3 above)
Comment (dhruska) [reply packages/core/src/utilities.ts:56]: It seems that, at least on the TS side, `StyleSheet.flatten` expects an array of objects always ((docs)[https://reactnative.dev/docs/stylesheet#flatten]) - so maybe what I can do is enforce the type of `baseStyles` on this function to take an array of objects, so we know that the styles aren't already flattened.

Comment (dhruska) [reply packages/core/src/utilities.ts:56]: 245a9a8

**Review by peterpme — APPROVED** [review]

**Pull request merged**